### PR TITLE
fix: enable testifylint on `staging/src/k8s.io/client-go`

### PR DESCRIPTION
**Pull request by mmorel-35**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This fixes rules from testifylint on `staging/src/k8s.io/client-go` package

### CODE REVIEW

**Review by Octopusjust — COMMENTED** [review]
/ok-to-test

**Review by Octopusjust — COMMENTED** [review]
/test pull-kubernetes-integration

**Review by krzyzacy — APPROVED** [review]

**Review by krzyzacy — APPROVED** [review]
/lgtm

**Review by krzyzacy — APPROVED** [review]
/lgtm

**Comment by BenTheElder** [comment]
/test pull-kubernetes-e2e-kind
https://github.com/kubernetes/kubernetes/issues/126401

File: staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery_test.go [hunk 1]
```diff
@@ -193,28 +193,26 @@ func TestOpenAPIDiskCache(t *testing.T) {
...
 
 				client.Invalidate()
 
 				// Refetch the schema from a new openapi client to try to force a new
 				// http request
 				newPaths, err := client.OpenAPIV3().Paths()
-				if !assert.NoError(t, err) {
-					continue
-				}
+				require.NoError(t, err)
```
Comment (thockin) [inline staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery_test.go:211]: Is this a correct conversion? The old logic looks like it flags an error and moves on to the next case without running further tests on this case. That does not appear to be what the new logic does.  IIUC it will abort this t.Run()?

File: staging/src/k8s.io/client-go/discovery/cached/memory/memcache_test.go [hunk 2]
```diff
@@ -431,38 +431,28 @@ func TestOpenAPIMemCache(t *testing.T) {
 		t.Run(contentType, func(t *testing.T) {
 			for k, v := range paths {
 				original, err := v.Schema(contentType)
-				if !assert.NoError(t, err) {
-					continue
-				}
+				require.NoError(t, err)
```
Comment (thockin) [inline staging/src/k8s.io/client-go/discovery/cached/memory/memcache_test.go:434]: Same comment as above on all of these cases

File: staging/src/k8s.io/client-go/discovery/cached/memory/memcache_test.go [hunk 3]
```diff
@@ -431,38 +431,28 @@ func TestOpenAPIMemCache(t *testing.T) {
...
 
 				schemaAgain, err := pathsAgain[k].Schema(contentType)
-				if !assert.NoError(t, err) {
-					continue
-				}
+				require.NoError(t, err)
 
-				assert.True(t, reflect.ValueOf(paths).Pointer() == reflect.ValueOf(pathsAgain).Pointer())
-				assert.True(t, reflect.ValueOf(original).Pointer() == reflect.ValueOf(schemaAgain).Pointer())
+				assert.Equal(t, reflect.ValueOf(paths).Pointer(), reflect.ValueOf(pathsAgain).Pointer())
```
Comment (thockin) [inline staging/src/k8s.io/client-go/discovery/cached/memory/memcache_test.go:442]: Is there any subtle behavior that Equal() implies that == does not?  ISTR some trap here, but cannot remember the details.

File: staging/src/k8s.io/client-go/kubernetes/fake/clientset_generated_test.go [hunk 4]
```diff
@@ -88,9 +90,8 @@ func TestManagedFieldClientset(t *testing.T) {
 	_, err = client.CoreV1().ConfigMaps("default").Apply(context.Background(),
 		v1ac.ConfigMap(name, namespace).WithData(map[string]string{"k1": "xyz"}),
 		meta_v1.ApplyOptions{FieldManager: "test-manager-2"})
-	if assert.Error(t, err) {
-		assert.Equal(t, "Apply failed with 1 conflict: conflict with \"test-manager-1\": .data.k1", err.Error())
-	}
+	require.Error(t, err)
```
Comment (thockin) [inline staging/src/k8s.io/client-go/kubernetes/fake/clientset_generated_test.go:93]: IIUC this will not run the next tests, where it used to?  Is that the intent?  The difference between `require` and `assert` is not as obvious as `Errorf` vs. `Fatalf`

File: staging/src/k8s.io/client-go/scale/client_test.go [hunk 5]
```diff
@@ -274,9 +275,7 @@ func TestGetScale(t *testing.T) {
 
 	for _, groupResource := range groupResources {
 		scale, err := scaleClient.Scales("default").Get(context.TODO(), groupResource, "foo", metav1.GetOptions{})
-		if !assert.NoError(t, err, "should have been able to fetch a scale for %s", groupResource.String()) {
-			continue
-		}
+		require.NoError(t, err, "should have been able to fetch a scale for %s", groupResource.String())
```
Comment (thockin) [inline staging/src/k8s.io/client-go/scale/client_test.go:278]: Same as above. If you think converting assert to require is correct, that should be a different PR because it is an actual behavior change

File: staging/src/k8s.io/client-go/scale/client_test.go [hunk 6]
```diff
@@ -302,9 +301,7 @@ func TestUpdateScale(t *testing.T) {
 
 	for _, groupResource := range groupResources {
 		scale, err := scaleClient.Scales("default").Update(context.TODO(), groupResource, expectedScale, metav1.UpdateOptions{})
-		if !assert.NoError(t, err, "should have been able to fetch a scale for %s", groupResource.String()) {
-			continue
-		}
+		require.NoError(t, err, "should have been able to fetch a scale for %s", groupResource.String())
```
Comment (thockin) [inline staging/src/k8s.io/client-go/scale/client_test.go:304]: Same here and elsewhere. I won't flag more

File: staging/src/k8s.io/client-go/testing/fixture_test.go [hunk 7]
```diff
@@ -277,7 +279,7 @@ func TestPatchWithMissingObject(t *testing.T) {
 	handled, node, err := reaction(action)
 	assert.True(t, handled)
 	assert.Nil(t, node)
-	assert.EqualError(t, err, `nodes "node-1" not found`)
+	require.EqualError(t, err, `nodes "node-1" not found`)
```
Comment (thockin) [inline staging/src/k8s.io/client-go/testing/fixture_test.go:282]: This is sort of a pointless change isn't it?

File: staging/src/k8s.io/client-go/testing/fixture_test.go [hunk 8]
```diff
@@ -429,33 +430,33 @@ func TestGetWithExactMatch(t *testing.T) {
 	nodeResource := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "node"}
 	node, gvr := constructObject(nodeResource, "node", "")
 
-	assert.Nil(t, o.Add(node))
+	require.NoError(t, o.Add(node))
```
Comment (thockin) [inline staging/src/k8s.io/client-go/testing/fixture_test.go:433]: Converting `assert.Nil` to `assert.NoError` is obvious.  Converting it to `require` means I have to read the rest of the test.  Better to do that as a separate PR

**Review by thockin — COMMENTED** [review]
It would be better if the conversions were equivalent in all cases. And then separately, call out places where assert should have been require, or vice versa
